Update caching docs (#2119)
codemagic-ci-cd/codemagic-docs: content/knowledge-codemagic/caching.md

  - /flutter-configuration/dependency-caching
---

> Caching can improve the efficiency of your build and deployment processes on Codemagic by reusing artifacts, such as packages, libraries, and compiled code.

Reviewer: When I read "artifacts" here, I immediately think of build artifacts, but all the alternatives I can think of, such as "components" or "dependencies", don't work better either.

Author: What if "artifacts" is indeed replaced with "components", but we elaborate a little what is meant by that. Such as
> components that are generated or fetched during the build, such as packages, ...

Reviewer: If that makes sense to you, go ahead.

Author: ```suggestion
Caching can improve the efficiency of your build and deployment processes on Codemagic by reusing components that are generated or fetched during the build, such as packages, libraries, and compiled code.
```

> The caching process begins when a successful build generates a cache. Codemagic stores this cache for a maximum of 14 days. After 14 days, the cache expires and is no longer accessible for subsequent builds. When this occurs, your workflow will revert to building without cache, which means dependencies and artifacts need to be fetched and generated again. If a build successfully completes without cache, Codemagic will automatically generate a new cache based on the build's output. This new cache is then uploaded and can be used for the next 14 days, following the same process.

Reviewer: Your version is fine too but I felt some useful stuff could be moved earlier in the paragraph.

```suggestion
When you have enabled caching for a workflow, Codemagic will automatically generate a cache based on the output of the first successful build. This cache is stored for a maximum of 14 days. After 14 days, the cache expires and is no longer accessible for subsequent builds. When this occurs, your workflow will fetch all dependencies and artifacts again and generate a new cache. This new cache is then uploaded and can be used for the next 14 days, following the same process.
```

## Cache usage limits

Maximum cache size is limited to

> - **10GB** per workflow for team account, and 

Reviewer: ```suggestion
- **10GB** per workflow for teams, and 
```
